G1ANT.Addon.Azure/Structures/AzureCredentialContainerStructure.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using G1ANT.Language;
+
+namespace G1ANT.Addon.Azure.Structures
+{
+    [Structure(Name = "azurecredentialcontainer", Tooltip = "This structure allows to use Azure Key Vault")]
+    class AzureCredentialContainerStructure : StructureTyped<PasswordStructure>
+    {
+        private string clientSecret;
+        private string clientId;
+        private Uri keyVaultUri;
+
+        public AzureCredentialContainerStructure(object value, string format = null, AbstractScripter scripter = null) : base(value, format, scripter) { }
+
+        public AzureCredentialContainerStructure(string clientSecret, string clientId, Uri keyVaultUri, string format = ""): base(null, format, null)
+        {
+            this.clientSecret = clientSecret;
+            this.clientId = clientId;
+            this.keyVaultUri = keyVaultUri;
+        }
+
+        public override Structure Get(string index = null)
```

Comment:
Default = null, but throws ArgumentException with default value? ;)

Reply:
Usage of this structure without an index does not make sense

Reply:
@pwitusik

Reply:
Yes, user has to pass the index to retrieve the value from the Key Vault